src/object/srv_obj.c

```diff
@@ -2273,10 +2273,12 @@ ds_obj_tgt_update_handler(crt_rpc_t *rpc)
 			rc = vos_dtx_commit(ioc.ioc_vos_coh,
 					    orw->orw_dti_cos.ca_arrays,
 					    orw->orw_dti_cos.ca_count, NULL);
-			if (rc < orw->orw_dti_cos.ca_count) {
-				D_ERROR(DF_UOID": Failed to DTX CoS commit "DF_RC".\n",
-					DP_UOID(orw->orw_oid), DP_RC(rc));
-				D_GOTO(out, rc);
+			if (rc < 0) {
+				D_WARN(DF_UOID": Failed to DTX CoS commit "DF_RC".\n",
+				       DP_UOID(orw->orw_oid), DP_RC(rc));
+			} else if (rc < orw->orw_dti_cos.ca_count) {
+				D_WARN(DF_UOID": Incomplete DTX CoS commit "DF_RC".\n",
+				       DP_UOID(orw->orw_oid), DP_RC(rc));
```

Comment:
You can't use DF_RC/DP_RC here if rc isn't a -DER_* error code, just use %d.

Reply:
Thank you very much! Just updated the code with %d.